inference-engine/src/cldnn_engine/cldnn_infer_request.cpp

```diff
@@ -891,39 +845,52 @@ void CLDNNInferRequest::InferImpl() {
     // execute input pre-processing.
     execDataPreprocessing(_inputs, true);  // "true" stands for serial preprocessing in case of OpenMP
 
-    for (auto &item : _inputs) {
-        std::string name = item.first;
-        Blob::Ptr inputBlob = item.second;
+    if (m_graph->GetMaxDynamicBatchSize() > 1) {
+        execAndParseDyn();
+        return;
+    }
 
-        if (m_graph->GetMaxDynamicBatchSize() > 1) {
-            PrepareInputDyn(name, *inputBlob);
-        } else {
-            auto nv12_ptr = inputBlob->as<NV12Blob>();
-            auto batched_ptr = inputBlob->as<BatchedBlob>();
-
-            if (nv12_ptr != nullptr || batched_ptr != nullptr) {
-                // special case for NV12 input blob
-                int num_blobs = batched_ptr != nullptr ? batched_ptr->size() : 1;
-                for (auto i = 0; i < num_blobs; i++) {
-                    if (batched_ptr != nullptr)
-                        nv12_ptr = getNV12BlobOrException(batched_ptr, i);
-
-                    PrepareInput(name + "_Y" + std::to_string(i), *nv12_ptr->y());
-                    PrepareInput(name + "_UV" + std::to_string(i), *nv12_ptr->uv());
+    // try locking stream infer mutex
+    auto mutex_ptr = m_graph->get_mutex();
+    mutex_ptr->lock();
```

Comment:
1. unlock is called in another function - it's pretty bad.
2. use std::lock_guard (RAII) instead of explicit lock/unlock calls.

Reply:
Fixed